proof/access-control/CNode_AC.thy

```diff
@@ -1176,13 +1177,19 @@ lemma sts_respects_restart_ep:
   apply (simp add: set_thread_state_def)
   apply (wpsimp wp: set_object_wp)
   apply (erule integrity_trans)
-  apply (clarsimp simp: integrity_def obj_at_def st_tcb_at_def get_tcb_def integrity_asids_kh_upds)
+  apply (clarsimp simp: integrity_def obj_at_def st_tcb_at_def get_tcb_def integrity_arch_kh_upds)
   apply (rule_tac tro_tcb_restart [OF refl refl])
     apply (fastforce dest!: get_tcb_SomeD)
    apply (fastforce dest!: get_tcb_SomeD)
   apply simp
   done
 
+(* FIXME AC: move *)
+lemma opt_map_upd_Some_unfolded[simp]:
+  "(\<lambda>x. if x = p then Some v else f x) |> g =
+   (\<lambda>x. if x = p then g v else (f |> g) x)"
```

Comment:
This is probably moved from elsewhere -- it'd make sense to put it into the option reader monad theory if we're touching it already anyway. I don't think it should be globally `[simp]`, though, at least it's not clear to me that this direction is always the one you'd want. We could do a `declare` simp here and move just the lemma.